### PR TITLE
feat: Add -lponly flag to export sub-command

### PR DESCRIPTION
**Pull request by ayang64**
When applied, this patch will add the -lponly flag to the export command
which instructs influx_inspect to only output line protocol without
comments and other out-of-band data.

Addresses: https://github.com/influxdata/influxdb/issues/19367

Describe your proposed changes here.

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [ ] Tests pass

### CODE REVIEW

File: cmd/influx_inspect/export/export.go [hunk 1]
```diff
@@ -209,42 +275,27 @@ func (cmd *Command) write() error {
...
-			fmt.Fprintln(cmd.Stdout, "complete.")
-		}
-		if _, ok := cmd.walFiles[key]; ok {
-			fmt.Fprintf(cmd.Stdout, "writing out wal file data for %s...", key)
-			if err := cmd.writeWALFiles(w, cmd.walFiles[key], key); err != nil {
-				return err
-			}
-			fmt.Fprintln(cmd.Stdout, "complete.")
+	// mw is our "meta writer" -- the io.Writer to which meta/out-of-band data
+	// like comments will be sent.  If the lponly flag is set, we mw will be
```
Comment (dgnorton) [inline cmd/influx_inspect/export/export.go:279]: nit: `we mw` should be `the mw`

File: cmd/influx_inspect/export/export.go [hunk 2]
```diff
@@ -209,42 +275,27 @@ func (cmd *Command) write() error {
...
+	// data..
+	//
+	mw := func() io.Writer {
+		if cmd.lponly {
+			return ioutil.Discard
 		}
-	}
-	return nil
+		return w
+	}()
```
Comment (dgnorton) [inline cmd/influx_inspect/export/export.go]: Saves a couple lines of code and simpler...
```
mw := w
if cmd.lponly {
   mw = ioutil.Discard
}
```

File: cmd/influx_inspect/export/export.go [hunk 3]
```diff
@@ -187,64 +191,111 @@ func (cmd *Command) walkWALFiles() error {
...
+	// Otherwise, mw is set to the same writer as the actual DDL and line
+	// protocol DML which will cause the comments to be intermixed with the
+	// data..
+	//
+	mw := func() io.Writer {
+		if cmd.lponly {
+			return ioutil.Discard
+		}
+		return w
+	}()
```
Comment (dgnorton) [inline cmd/influx_inspect/export/export.go:292]: Saves a couple lines of code and simpler...
```
mw := w
if cmd.lponly {
   mw = ioutil.Discard
}
```

**Review by dgnorton — CHANGES_REQUESTED** [review]
Mostly looks good. Couple small things.

Have you tested this to confirm it generates just line protocol?

**Comment by ayang64** [comment]
> Mostly looks good. Couple small things.
> 
> Have you tested this to confirm it generates just line protocol?

Yup.  Works for me.

File: cmd/influx_inspect/export/export.go (hunk 2 above)
Comment (ayang64) [reply cmd/influx_inspect/export/export.go]: Done.  Just so you know, I did think about this.  My intention here is that I feel like with:

```
mw := w
if cmd.lponly {
  mw = ioutil.Discard
}
```
as you read it, you have to track the value of `mw` through the condition and the variable name is intermixed with the logic for assigning itself.   Personally, I like assigning `mw` only once and keeping all of the logic that dictates is value separate; it feels easier to read to me.  So I think about it -- if maybe came to a different conclusion.

File: cmd/influx_inspect/export/export.go (hunk 1 above)
Comment (ayang64) [reply cmd/influx_inspect/export/export.go]: d'oh -- fixed!

**Review by dgnorton — APPROVED** [review]

**Pull request merged**